auth/realm/token/src/main/java/org/wildfly/security/auth/realm/token/TokenSecurityRealm.java

```diff
@@ -183,19 +183,19 @@ public SupportLevel getCredentialAcquireSupport(Class<? extends Credential> cred
 
         @Override
         public SupportLevel getEvidenceVerifySupport(Class<? extends Evidence> evidenceType, String algorithmName) throws RealmUnavailableException {
-            if (exists() && isBearerTokenEvidence(evidenceType)) {
+            if (isBearerTokenEvidence(evidenceType)) {
                 return SupportLevel.SUPPORTED;
             }
 
             return SupportLevel.UNSUPPORTED;
         }
 
-        private Attributes getClaims() throws RealmUnavailableException {
-            if (this.claims == null) {
-                this.claims = validateToken(this.evidence);
-            }
+        private void setClaims(Attributes claims) throws RealmUnavailableException {
+            this.claims = claims;
+        }
 
-            return this.claims;
+        private Attributes getClaims() throws RealmUnavailableException {
+            return validateToken(this.evidence);
```

Comment:
We'll still need the `if` statement here in case `this.claims` is `null`, i.e.,

```
if (this.claims == null) {
    validateToken(this.evidence);
}
return this.claims;
```

Just meant that the line `this.claims = validateToken(this.evidence);` could be simplified to just `validateToken(this.evidence);`.

Reply:
@fjuma Fixed. Thank you